web/src/main/java/org/cbioportal/web/GenesetDataController.java

```diff
@@ -36,27 +36,39 @@
 
     @PreAuthorize("hasPermission(#geneticProfileId, 'GeneticProfileId', 'read')")
     @RequestMapping(value = "/genetic-profiles/{geneticProfileId}/geneset-genetic-data/fetch", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE,
-            produces = MediaType.APPLICATION_JSON_VALUE)
+        produces = MediaType.APPLICATION_JSON_VALUE)
     @ApiOperation("Fetch gene set \"genetic data\" items (gene set scores) by profile Id, gene set ids and sample ids")
     public ResponseEntity<List<GenesetMolecularData>> fetchGeneticDataItems(
-            @ApiParam(required = true, value = "Genetic profile ID, e.g. gbm_tcga_gsva_scores")
-            @PathVariable String geneticProfileId,
-            @ApiParam(required = true, value = "Search criteria to return the values for a given set of samples and gene set items. "
-                    + "genesetIds: The list of identifiers for the gene sets of interest, e.g. HINATA_NFKB_MATRIX. "
-                    + "Use one of these if you want to specify a subset of samples:"
-                    + "(1) sampleListId: Identifier of pre-defined sample list with samples to query, e.g. brca_tcga_all "
-                    + "or (2) sampleIds: custom list of samples or patients to query, e.g. TCGA-BH-A1EO-01, TCGA-AR-A1AR-01")
-            @RequestBody GenesetDataFilterCriteria genesetDataFilterCriteria) throws MolecularProfileNotFoundException, SampleListNotFoundException {
-
-        if (genesetDataFilterCriteria.getSampleListId() != null && genesetDataFilterCriteria.getSampleListId().trim().length() > 0) {
+        @ApiParam(required = true, value = "Genetic profile ID, e.g. gbm_tcga_gsva_scores")
+        @PathVariable String geneticProfileId,
+        @ApiParam(required = true, value = "Search criteria to return the values for a given set of samples and gene set items. "
+            + "genesetIds: The list of identifiers for the gene sets of interest, e.g. HINATA_NFKB_MATRIX. "
+            + "Use one of these if you want to specify a subset of samples:"
+            + "(1) sampleListId: Identifier of pre-defined sample list with samples to query, e.g. brca_tcga_all "
+            + "or (2) sampleIds: custom list of samples or patients to query, e.g. TCGA-BH-A1EO-01, TCGA-AR-A1AR-01")
+        @RequestBody GenesetDataFilterCriteria genesetDataFilterCriteria) throws MolecularProfileNotFoundException, SampleListNotFoundException {
+
+        if (sampleListIdExists(genesetDataFilterCriteria)) {
```

Comment:
This is a place where I think Clean Code is wrong. Pulling the predicate out into its own method when it's used in exactly one spot, and it's design is already clear is not worth it to me. I think this increases the mental effort needed to read this code, as you have to navigate to `sampleListIdExists` to understand the logic, and the logic isn't fully covered by the method name `sampleListIdExists`. I would change this back.